plugins/flytekit-papermill/flytekitplugins/papermill/task.py

```diff
@@ -256,7 +259,8 @@ def execute(self, **kwargs) -> Any:
         """
         logger.info(f"Hijacking the call for task-type {self.task_type}, to call notebook.")
         # Execute Notebook via Papermill.
-        pm.execute_notebook(self._notebook_path, self.output_notebook_path, parameters=kwargs, log_output=self._stream_logs)  # type: ignore
+        seralized_kwargs = serialize_inputs(**kwargs)
```

Comment:
It would be nice to only serialize the datatypes that need to be serialized. I wasn't sure how to check for that.

Reply:
something like:
```python
for k, v in kwargs.items():
  if is_instance(v, StructuredDataset):
    kwargs[k] = serialize_inputs(v)
```